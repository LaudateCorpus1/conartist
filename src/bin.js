#!/usr/bin/env node

const meow = require("meow");
const { getConfig, process } = require("./process");

const cli = meow(`
  Usage
    conartist
`);

(async () => {
  await process(await getConfig());
})();
