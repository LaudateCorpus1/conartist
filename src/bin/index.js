#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const yargs = require('yargs');

const { argv } = yargs;
const cmd = argv._.join(path.delimiter) || 'sync';

if (fs.existsSync(path.join(__dirname, '..', 'node', 'bin', `${cmd}.js`))) {
  require(`../node/bin/${cmd}`)(argv);
} else {
  throw new Error(`Command not found: ${cmd}`);
}
