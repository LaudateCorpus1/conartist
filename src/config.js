const cosmiconfig = require("cosmiconfig");

async function getConfig() {
  const config = await cosmiconfig("conartist").search();
  return config ? config.config : {};
}

module.exports = {
  getConfig
};
