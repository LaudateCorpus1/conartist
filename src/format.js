const prettier = require("prettier");

function formatCode(code) {
  return prettier.format(code, { parser: "babel" });
}

function formatJson(json) {
  json = typeof json === "string" ? JSON.parse(json) : json;
  return JSON.stringify(json, null, 2);
}

module.exports = {
  formatCode,
  formatJson
};
