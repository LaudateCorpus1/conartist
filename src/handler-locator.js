const path = require("path");

const { array, js, json, string } = require("./handler");

let currentHandlerLocator;

function getHandlerLocator() {
  return currentHandlerLocator;
}

function setHandlerLocator(newHandlerLocator) {
  const oldHandlerLocator = currentHandlerLocator;
  currentHandlerLocator = newHandlerLocator;
  return oldHandlerLocator;
}

setHandlerLocator(function(key, val) {
  const basename = path.basename(key);
  const extname = path.extname(key);
  const type = typeof val;

  if (basename === ".babelrc") return json;
  if (basename === ".eslintrc") return json;
  if (extname === ".js") return js;
  if (type === "object") return json;

  return string;
});

module.exports = {
  getHandlerLocator,
  setHandlerLocator
};
