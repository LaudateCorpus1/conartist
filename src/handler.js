const fs = require("fs-extra");
const isFunction = require("lodash/isFunction");
const loMerge = require("lodash/merge");
const loUniq = require("lodash/uniq");
const minimatch = require("minimatch");
const path = require("path");
const stripIndent = require("strip-indent");
const { formatCode, formatJson, formatMd } = require("./format");

let currentHandler;

function merge(...args) {
  return loMerge({}, ...args);
}

async function readIfExists(file) {
  return (await fs.exists(file)) ? await fs.readFile(file) : null;
}

async function requireIfExists(file) {
  return (await fs.exists(file)) ? require(file) : null;
}

async function handleArray(file) {
  let data;
  const curr = ((await readIfExists(file.name)) || "").split("\n");
  data = file.data.concat(curr);
  data = loUniq(data);
  return data.filter(Boolean).join("\n");
}

async function handleJs(file) {
  const curr = (await readIfExists(file.name)) || file.data;
  const data = file.overwrite ? file.data : curr;
  return formatCode(data);
}

async function handleJson(file) {
  const curr = (await requireIfExists(file.name)) || file.data;
  const data = file.overwrite
    ? file.data
    : file.merge
    ? merge(curr, file.data)
    : curr;
  return JSON.stringify(data, null, 2);
}

async function handleMd(file) {
  return formatMd(await handleString(file));
}

async function handleString(file) {
  const curr = (await readIfExists(file.name)) || file.data;
  const data = file.overwrite ? file.data : curr;
  return stripIndent(data).trim();
}

const mapExtname = {
  js: handleJs,
  jsx: handleJs,
  json: handleJson,
  md: handleMd
};

const mapType = {
  object: handleJson
};

function getType(file) {
  let type;

  if (isFunction(file.data)) {
    return file.data;
  }

  if (isFunction(file.type)) {
    return file.type;
  }

  type =
    file.type ||
    path
      .extname(file.name)
      .substring(1)
      .toLowerCase();
  if (type in mapExtname) {
    return mapExtname[type];
  }

  type = typeof file.data;
  if (type in mapType) {
    return mapType[type];
  }

  return handleString;
}

async function handler(file) {
  return await getType(file)(file);
}

module.exports = {
  handler
};
