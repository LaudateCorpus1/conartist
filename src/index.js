const { config } = require('./config');
const { formatCode, formatJson } = require('./format');
const { js, json, string } = require('./handler');
const { getHandlerLocator, setHandlerLocator } = require('./handler-locator');
const { load, raw, resolve } = require('./load');
const { sync } = require('./sync');

module.exports = {
  getHandlerLocator,
  setHandlerLocator,
  config,
  formatCode,
  formatJson,
  js,
  json,
  load,
  raw,
  resolve,
  string,
  sync
};
