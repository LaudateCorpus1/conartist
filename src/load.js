const fs = require('fs');
const path = require('path');

function load(filepath) {
  const resolved = resolve(filepath);
  return resolved && require(resolved);
}

function raw(filepath) {
  const resolved = resolve(filepath);
  return resolved && fs.readFileSync(resolved).toString();
}

function resolve(filepath) {
  try {
    return require.resolve(path.join(process.cwd(), filepath));
  } catch (e) {}
}

module.exports = {
  load,
  raw,
  resolve
};
