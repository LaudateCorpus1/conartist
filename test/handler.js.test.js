import { js } from '../src';

jest.mock('../src/load');
const { resolve } = require('../src/load');
resolve.mockImplementation(() => 'path/to/conartist.js');

describe('handler.js', () => {
  it('function', () => {
    expect(js()(() => {}, 'testfile')).toMatchSnapshot();
  });
  it('object', () => {
    expect(js()({}, 'testfile')).toMatchSnapshot();
  });
});
