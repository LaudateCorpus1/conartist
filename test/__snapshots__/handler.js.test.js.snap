// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handler.js function 1`] = `
"// This file is autogenerated to retain the scope the JS is defined in.
module.exports = require('./path/to/conartist.js')['testfile'].data();
"
`;

exports[`handler.js object 1`] = `
"// This file is autogenerated to retain the scope the JS is defined in.
module.exports = require('./path/to/conartist.js')['testfile'].data;
"
`;
